horace_core/sqw/PixelData/@PixelData/set_data.m

```diff
@@ -0,0 +1,112 @@
+function set_data(obj, pix_fields, data, varargin)
+%SET_PIXELS Update the data on the given pixel data fields
+%
+% The number of columns in 'data' must be equal to the number of fields in
+% 'pix_fields'. The number of rows in 'data' must be equal to the number of
+% elements in 'abs_pix_indices'.
+%
+% Examples:
+% ---------
+%
+% Set the first 100 pixels' signal and variance to zero
+%   >> set_data({'signal', 'variance'}, zeros(2, 100), 1:100);
+%
+% Input:
+% ------
+% pix_fields       The name of a field, or a cell array of field names.
+% data             The data with which to set the given fields.
+% abs_pix_indices  The indices to set data on. If not specified all indices are
+%                  updated and 'size(data, 2)' must equal to obj.num_pixels.
+%
+NO_INPUT_INDICES = -1;
+
+[pix_fields, abs_pix_indices] = parse_args(obj, pix_fields, data, varargin{:});
+
+field_indices = cell2mat(obj.FIELD_INDEX_MAP_.values(pix_fields));
+
+if obj.is_file_backed_()
+    base_pg_size = obj.base_page_size;
+
+    if abs_pix_indices == NO_INPUT_INDICES
+        first_required_page = 1;
+    else
+        first_required_page = ceil(min(abs_pix_indices)/base_pg_size);
+    end
+    obj.move_to_page(first_required_page);
+
+    set_page_data(obj, field_indices, data, abs_pix_indices);
+    while obj.has_more()
+        obj.advance();
+        set_page_data(obj, field_indices, data, abs_pix_indices);
+    end
+else
+    if abs_pix_indices == NO_INPUT_INDICES
+        obj.data_(field_indices, 1:end) = data;
+    else
+        obj.data_(field_indices, abs_pix_indices) = data;
+    end
+end
+
+end  % function
+
+
+% -----------------------------------------------------------------------------
+function [pix_fields, abs_pix_indices] = parse_args(obj, pix_fields, data, varargin)
+    NO_INPUT_INDICES = -1;
+
+    validateattributes(pix_fields, {'cell', 'char', 'string'}, {'nonempty'});
+    validateattributes(data, {'numeric'}, {});
+
+    parser = inputParser();
+    parser.addOptional( ...
+        'abs_pix_indices', ...
+        NO_INPUT_INDICES, ...
+        @(x) validateattributes(x, {'numeric', 'logical'}, {'integer', 'nonnegative'}) ...
```

Comment:
Question more for my own sake, does the check function only apply IF the argument is present?